src/components/TableComplex/RemoveDeviceDialog.react.js

```diff
@@ -144,6 +144,7 @@ class RemoveDeviceDialog extends Component {
                   verticalAlign: 'middle',
                 }}
                 disabled={!this.state.correctName}
+                ref={el = this.removeDeviceButton = el}
```

Comment:
I don't think keeping all those rules within the state is a good idea since it is not accessed globally. I have defined them as objects within the ```life-cycle hook``` method and assigned correspondingly.